### ISSUE TITLE
modify_dn() delete_old_dn keyword broken in mock connections

### ISSUE THREAD

**Issue by kpinc** [cannatag/ldap3#810]
[delete_old_dn.patch.zip](https://github.com/cannatag/ldap3/files/4423267/delete_old_dn.patch.zip)

The delete_old_dn flag for modify_dn() is implemented wrong in
the mock connection code.

The flag _should_ control whether or not the attribute(s?)
changed between the new dn and the old dn are updated in the
entry itself.  See: https://ldap.com/the-ldap-modify-dn-operation/

The code instead uses the flag to control whether or not the old
entry is removed from the DIT.

An entirely untested patch is attached.

Note that I am unclear on what the LDAP specification says should
happen to attributes which appear in the old dn but do not appear
in the new dn, when delete_old_dn is True.  Conceivably these
attributes should be removed from the entry.  The code of my
patch does not do this.

**Comment by cannatag** [comment]
Thanks, LDAP specification are poor on many side effects... Will have a look at yout patch soon.

**Comment by cannatag** [comment]
Hi, I've updated the code in dev, can you check if it works as expected?

thanks,
Giovanni

**Comment by kpinc** [comment]
On Mon, 06 Jul 2020 08:24:35 -0700
Giovanni Cannata <notifications@github.com> wrote:

> Hi, I've updated the code in dev, can you check if it works as
> expected?

Hi,

I'm sorry but I have no expectations.  I wrote some code to use
it and found it didn't work according to the specification.
I whacked together a patch that seemed to work, sent it to you,
and later abandoned what I was working on without ever developing
a real test case.

So I don't really have anything to test against at this point.

Apologies for being un-helpful.

Note that the patch I sent you was ugly -- it left old
code that was simply commented out.
I think it'd be appropriate if there was another patch
which removes the commented out code.

Also, there's no test case.  I'd expect that testLDIF-change.py,
function test_modify_dn_request_to_ldif() should be updated
to test for what's expected.  But I'm not familiar with
your test framework so can't quickly send in a patch for that.

Regards,

Karl <kop@karlpinc.com>
Free Software:  "You don't pay back, you pay forward."
                 -- Robert A. Heinlein
